telegram/_bot.py

```diff
@@ -442,13 +442,51 @@ async def _send_message(
         allow_sending_without_reply: ODVInput[bool] = DEFAULT_NONE,
         protect_content: ODVInput[bool] = DEFAULT_NONE,
         message_thread_id: int = None,
+        caption: str = None,
+        caption_entities: Union[List["MessageEntity"], Tuple["MessageEntity", ...]] = None,
         *,
         read_timeout: ODVInput[float] = DEFAULT_NONE,
         write_timeout: ODVInput[float] = DEFAULT_NONE,
         connect_timeout: ODVInput[float] = DEFAULT_NONE,
         pool_timeout: ODVInput[float] = DEFAULT_NONE,
         api_kwargs: JSONDict = None,
     ) -> Union[bool, Message]:
+        """Protected method to send messages of any type. Currently calls :meth:`_post`.
+
+        Note:
+            To reduce repetition (`if`-clauses) in public methods, the parameters
+            :paramref:`caption`, :paramref:`caption_entities`, :paramref:`message_thread_id`,
+            :paramref:`reply_markup`, and :paramref:`reply_to_message_id` are included
+            separately instead of them being passed in :paramref:`data`
+
+        Args:
+            endpoint (:obj:`str`): Telegram API endpoint
+            data (:class:`telegram._utils.types.JSONDict`): Data to be sent
+            reply_to_message_id (:obj:`int`, optional): |reply_to_msg_id|
+            disable_notification (:obj:`bool`, optional): |disable_notification|
+            reply_markup (:class:`ReplyMarkup`, optional): Additional interface options.
+                An object for an inline keyboard, custom reply keyboard, instructions to remove
+                reply keyboard or to force a reply from the user.
+            allow_sending_without_reply (:obj:`bool`, optional): |allow_sending_without_reply|
+            protect_content (:obj:`bool`, optional): |protect_content|
+
+                .. versionadded:: 13.10
+            message_thread_id (:obj:`int`, optional): |message_thread_id_arg|
+
+                .. versionadded:: 20.0
+            caption (:obj:`str`, optional): Media caption (may also be used when resending photos
+                by file_id), 0-:tg-const:`telegram.constants.MessageLimit.CAPTION_LENGTH`
+                characters after entities parsing.
+            caption_entities (List[:class:`telegram.MessageEntity`], optional): |caption_entities|
+
+        Returns:
+            :class:`telegram.Message`: On success, depending on a type of operation, the message
+                or :obj:`True` is returned.
+
+        Raises:
+            :class:`telegram.error.TelegramError`
+
+        """
```

Comment:
```suggestion
        """Protected method to send or edit messages of any type. 

        It is here to reduce repetition of if-else closes in the different bot methods, i.e. this method takes care of adding its parameters to `data` if appropriate.

        Depending on the bot method, returns either `True` or the message.
        """
```

Sorry, I should have been more explicit in my previous comment. 

> give the method an explanation about why it's there.

What I'm looking for here is mostly a short info about what this method does. Since it's a description for the complete method and not only for parts of its body, a docstring seems more approriate to me than a `#` style comment. Nonetheless, because sphinx doesn't render docstrings of protected methods, it *is* basically a comment, so correct sphinx/rst-syntax is not strictly needed.
Because the parameters are already well-documented all over `_bot.py`, I don't see a need to repeat this here. Finally the `Note` that you added is the most integral part of the docstring :)

Reply:
Done